starfish/core/image/_filter/reduce.py

```diff
@@ -0,0 +1,123 @@
+import warnings
+from copy import deepcopy
+from typing import Callable, Iterable, Optional, Union
+
+import numpy as np
+
+from starfish.core.imagestack.imagestack import ImageStack
+from starfish.core.types import Axes
+from starfish.core.util import click
+from ._base import FilterAlgorithmBase
+
+
+class Reduce(FilterAlgorithmBase):
+    """
+    Reduces the dimensions of the ImageStack by applying a function
+    along one or more axes.
+
+    Parameters
+    ----------
+    dims : Axes
+        one or more Axes to project over
+    func : Union[str, Callable]
```

Comment:
why not accept just Callable?

Reply:
i like it. this is the more 'pythonic' way in scientific computing -- will be easier for numpy/pandas users to reason about i think.

Reply:
@dganguli You like what?  The original or my suggestion?

Reply:
I think it's nice to have the string option. I think a lot of users will find it annoying to need to import numpy.amax() and this component to perform a maximum intensity projection.

Reply:
I support the string option, just for ecosystem continuity -- users are used to passing function names that are resolved by getattr, as this formalism is used a lot in `scipy.spatial.distance` among other packages.